app/views/bulk_submission_forms/_form_fields.html.erb

```diff
@@ -1,26 +1,65 @@
 <%= form.govuk_error_summary %>
 
+<%= render partial: "error_summary_hidden" %>
+
 <%= form.govuk_fieldset legend: { text: t(".page_title"), tag: "h1", size: "xl" } do %>
   <div class="govuk-body govuk-!-padding-top-4"></div>
 
-  <!-- Non-JS fallback upload form -->
-  <div class="fallback no-script">
-    <%= form.govuk_file_field(
-          :uploaded_file,
-          name: :uploaded_file,
-          label: { text: t("generic.attach_file"), size: "m" },
-          hint: { text: t(".size_hint") },
-          classes: ["govuk-file-upload"]
-        ) %>
-
-    <%= form.govuk_submit(
-          t("generic.upload"),
-          id: "upload",
-          name: "commit",
-          value: "upload",
-          secondary: true,
-        ) %>
+  <% form_error_class = form.object.errors.any? ? "govuk-form-group--error" : "" %> <%# erblint:disable Rubocop %>
+
+  <!-- Dropzone upload form -->
+  <div class="dropzone__upload">
+    <div class="govuk-form-group script hidden <%= form_error_class %>" id="dropzone-form-group" aria-labelledby="dropzone-label">
+      <label class="govuk-label govuk-label--m" id="dropzone-label">
+          <%= t("generic.attach_file") %>
+      </label>
+
+      <div class="govuk-hint"><%= t(".size_hint") %></div>
+
+      <% if form.object.errors %>
```

Comment:
I think you might be able to remove this whole if block - I might be wrong but in the original JS I think this block was added to display errors directly from the form object. But here all the errors from the form object are returned as JSON and rendered via the dropzone error handle, is that right?

Reply:
thanks, I will take a look. we do pass the errors via JSON so if the summary and field level are being generated via the same JSON then hopefully this can go 🤞

Reply:
Turns out this block is necessary for an "edge" case where a user hits Save and continue without adding any file.

If no file is chosen then, while no-JS will add the "You must select a file" error to the summary _and file field_, the JS/Dropzone file field will not get the error added (because the file field is a dropzone file field that is not found by govuk form builder gem's handling) and **dropzone has not been invoked**, so its `on('error')` event never gets the chance to add it either. So this block patches that edge case.

on the plus side, the system tests caught this edge case explicitly even though my manual testing did not 😄

Reply:
Ahh okay, makes sense, thanks for investigating/explaining!